libcloud/utils/misc.py

```diff
@@ -282,3 +292,28 @@ def __repr__(self):
 
     def __str__(self):
         return str(self.__repr__())
+
+
+def retry(timeout, retry_delay, backoff, func, *args, **kwargs):
+    """
+    Retry decorator helps to handle common exception.
+    """
+    timeout = DEFAULT_TIMEOUT if timeout is None else timeout
```

Comment:
@jimbobhickville That's correct - you would either need to set the default value before the import or you need to explicitly specify the timeout argument.